iceoryx_posh/include/iceoryx_posh/popo/listener.hpp

```diff
@@ -84,15 +84,15 @@ class Listener
     /// @param[in] eventOrigin the object which will signal the event (the origin)
     /// @param[in] eventType enum required to specify the type of event inside of eventOrigin
     /// @param[in] eventCallback callback which will be executed concurrently when the event occurs. has to be created
-    /// with iox::popo::createEventCallback
+    /// with iox::popo::createNotificationCallback
     /// @return If an error occurs the enum packed inside an expected which describes the error.
     template <typename T,
               typename EventType,
               typename ContextDataType,
               typename = std::enable_if_t<std::is_enum<EventType>::value>>
     cxx::expected<ListenerError> attachEvent(T& eventOrigin,
                                              const EventType eventType,
-                                             const EventCallback<T, ContextDataType>& eventCallback) noexcept;
+                                             const NotificationCallback<T, ContextDataType>& eventCallback) noexcept;
```

Comment:
Should we rename this to `notificationCallback` too? If so, don't forget the doxygen comment.

Reply:
in this particular case I would say no since it is a callback which corresponds to an attached event. notification is for me the superset of event/state somehow and here we have an event notification with an event callback.